src/cclib/io/filereader.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Generic file reader and related tools"""
+
+
+class Reader(object):
+    """Abstract class for reader objects."""
```

Comment:
If this is really an abstract class, then use abc?

Reply:
abc? https://docs.python.org/3.6/library/abc.html?

Reply:
Yeah, if if it's really meant to be an abstract base class.

Reply:
I guess I'm not really sure. I'm looking for C++-style behavior.

Reply:
Could you be more specific what "C++-style behavior"?